### ISSUE TITLE
`pk;system front` command does not respect Discord message length limit and throws an error

### ISSUE THREAD

**Issue by spiralw** [xSke/PluralKit#360]

**Comment by spiralw** [comment]
Other commands have similar issues, such as group list / group card. Might be worth a wrapper to CreateMessage that does length checking?

**Comment by xSke** [comment]
A lot of the time we need to do more complex truncation logic instead of just cutting off the end, so not sure a generic wrapper would work :/

**Comment by spiralw** [comment]
True, but is it better to cut stuff incorrectly - possibly showing something wrong - or throw an internal error? (yes, the better option is to truncate it correctly, but clearly that doesn't always happen)

**Comment by xSke** [comment]
Yeah, that's fair.

**Comment by rladenson** [comment]
Maybe you could make the wrapper but not cut off access to CreateMessage and functions that have their own custom truncation code could bypass it?

**Comment by spiralw** [comment]
Ideally functions that have their own custom truncation code would respect Discord limits and not need to get cleaned up by the wrapper; the wrapper is essentially a no-op if whatever is calling CreateMessage isn't sending anything that's too long.

**Comment by rladenson** [comment]
I think that's what I meant? If it has its own truncation code it calls MakeMessage itself, if not it calls the wrapper.

**Comment by spiralw** [comment]
well, I don't want to blindly trust context-specific truncation code, and it would be *more* work to selectively apply it to only stuff that doesn't have its own checks. Checking the length twice doesn't hurt, I guess.
